lib/time/time.go

```diff
@@ -0,0 +1,548 @@
+type Time time.Time
+
+func newTime(thread *starlark.Thread, _ *starlark.Builtin, args starlark.Tuple, kwargs []starlark.Tuple) (starlark.Value, error) {
+	var (
+		year, month, day, hour, min, sec, nsec int
+		loc                                    string
+	)
+	if err := starlark.UnpackArgs("time", args, kwargs, "year", &year, "month", &month, "day", &day, "hour", &hour, "min", &min, "sec", &sec, "nsec", &nsec, "loc", &loc); err != nil {
+		return nil, err
+	}
+	location, err := time.LoadLocation(loc)
+	if err != nil {
+		return nil, err
+	}
+	return Time(time.Date(year, time.Month(month), day, hour, min, sec, nsec, location)), nil
+}
+
+// String implements the Stringer interface.
+func (t Time) String() string { return time.Time(t).String() }
+
+// Type returns "time.time".
+func (t Time) Type() string { return "time.time" }
+
+// Freeze renders time immutable. required by starlark.Value interface
+// because Time is already immutable this is a no-op.
+func (t Time) Freeze() {}
+
+// Hash returns a function of x such that Equals(x, y) => Hash(x) == Hash(y)
+// required by starlark.Value interface.
+func (t Time) Hash() (uint32, error) {
+	return starlark.MakeInt64(time.Time(t).UnixNano()).Hash()
+}
+
+// Truth returns the truth value of an object required by starlark.Value
+// interface.
+func (t Time) Truth() starlark.Bool { return starlark.Bool(time.Time(t).IsZero()) }
+
+// Attr gets a value for a string attribute, implementing dot expression support
+// in starklark. required by starlark.HasAttrs interface.
+func (t Time) Attr(name string) (starlark.Value, error) {
+	switch name {
+	case "year":
+		return starlark.MakeInt(time.Time(t).Year()), nil
+	case "month":
+		return starlark.MakeInt(int(time.Time(t).Month())), nil
+	case "day":
+		return starlark.MakeInt(time.Time(t).Day()), nil
+	case "hour":
+		return starlark.MakeInt(time.Time(t).Hour()), nil
+	case "minute":
+		return starlark.MakeInt(time.Time(t).Minute()), nil
+	case "second":
+		return starlark.MakeInt(time.Time(t).Second()), nil
+	case "nanosecond":
+		return starlark.MakeInt(time.Time(t).Nanosecond()), nil
+	case "unix":
+		return starlark.MakeInt64(time.Time(t).Unix()), nil
+	case "unix_nano":
+		return starlark.MakeInt64(time.Time(t).UnixNano()), nil
+	}
+	return builtinAttr(t, name, timeMethods)
+}
+
+// AttrNames lists available dot expression strings for time. required by
+// starlark.HasAttrs interface.
+func (t Time) AttrNames() []string {
+	return append(builtinAttrNames(timeMethods),
+		"year",
+		"month",
+		"day",
+		"hour",
+		"minute",
+		"second",
+		"nanosecond",
+		"unix",
+		"unix_nano",
+	)
+}
+
+// CompareSameType implements comparison of two Time values. required by
+// starlark.Comparable interface.
+func (t Time) CompareSameType(op syntax.Token, yV starlark.Value, depth int) (bool, error) {
+	x := time.Time(t)
+	y := time.Time(yV.(Time))
+	cp := 0
+	if x.Before(y) {
+		cp = -1
+	} else if x.After(y) {
+		cp = 1
+	}
+	return threeway(op, cp), nil
```

Comment:
I prefer doing the comparison directly here. But this is just my two cent...
```suggestion
	switch op {
￼	case syntax.EQL:
￼		return x.Equal(y), nil
￼	case syntax.NEQ:
￼		return !x.Equal(y), nil
￼	case syntax.LT:
￼		return x.Before(y), nil
￼	case syntax.LE:
￼		return !x.After(y), nil
￼	case syntax.GT:
￼		return x.After(y), nil
￼	case syntax.GE:
￼		return !x.Before(y), nil
￼	}

	return false, errors.New("operation not supported")
￼￼```